### PR TITLE
feat(common): add support for gateway api routes

### PR DESCRIPTION
**Pull request by andrasmaroy**
**Description of the change**

Adds support for [Gateway API Routes](https://gateway-api.sigs.k8s.io/) similar to ingresses currently.

**Benefits**

With this update people using Gateway API for networking are able to add routes for their services using this chart instead of having to add only the route for the service created by chart.

The idea was to handle this behavior similar to how Ingresses are handled currently by the chart.

**Possible drawbacks**

Non really that I can think of, one caveat is that the Gateway API uses CRDs, so the user would need to install those first before using them with this chart.

**Applicable issues**

N/A

**Additional information**

N/A

**Checklist** 
- [X] Title of the PR conforms to the [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/) standard.
- [X] Scope of the of the PR title contains the chart name.
- [X] Chart version in `Chart.yaml` has been bumped according to [Semantic Versioning](https://semver.org/).
- [X] Chart `artifacthub.io/changes` changelog annotation has been updated in `Chart.yaml`. See [Artifact Hub documentation](https://artifacthub.io/docs/topics/annotations/helm/#supported-annotations) for more info.
- [X] Variables have been documented in the `values.yaml` file.

### CODE REVIEW

File: charts/library/common/Chart.yaml [hunk 1]
```diff
@@ -2,7 +2,7 @@ apiVersion: v2
 name: common
 description: Function library for Helm charts
 type: library
-version: 0.2.2
+version: 0.3.0
 kubeVersion: ">=1.16.0-0"
```
Comment (onedr0p) [inline charts/library/common/Chart.yaml:6]:
```suggestion
kubeVersion: ">=1.19.0-0"
```

File: charts/other/app-template/Chart.yaml [hunk 2]
```diff
@@ -2,19 +2,19 @@
 apiVersion: v2
 description: A common powered chart template. This can be useful for small projects that don't have their own chart.
 name: app-template
-version: 0.2.2
+version: 0.3.0
```
Comment (onedr0p) [inline charts/other/app-template/Chart.yaml:5]: Should we also add in `kubeVersion` here or?

**Comment by onedr0p** [comment]
@andrasmaroy would you mind adding in some unit test? Please look at the other examples if you are not familiar with this.

**Comment by andrasmaroy** [comment]
> @andrasmaroy would you mind adding in some unit test? Please look at the other examples if you are not familiar with this.

Sure thing, though I most likely will only have time early next week.

File: charts/library/common/Chart.yaml [hunk 3]
```diff
@@ -2,8 +2,8 @@ apiVersion: v2
 name: common
 description: Function library for Helm charts
 type: library
-version: 0.3.0
-kubeVersion: ">=1.16.0-0"
+version: 0.4.0
+kubeVersion: ">=1.19.0-0"
```
Comment (bjw-s) [inline charts/library/common/Chart.yaml:6]: Having this minimum kubeVersion implies certain other functionality has also been changed/removed which is not the case yet. I have a major refactor/rewrite coming up to take care of this but until then I want to keep the k8s version compatibility intact.

File: charts/library/common/Chart.yaml [hunk 4]
```diff
@@ -2,8 +2,8 @@ apiVersion: v2
 name: common
 description: Function library for Helm charts
 type: library
-version: 0.3.0
-kubeVersion: ">=1.16.0-0"
+version: 0.4.0
```
Comment (bjw-s) [inline charts/library/common/Chart.yaml:5]: The `development` branch version has already been bumped, shouldn't require another version bump

File: charts/library/common/Chart.yaml [hunk 5]
```diff
@@ -14,4 +14,4 @@ maintainers:
 annotations:
   artifacthub.io/changes: |-
     - kind: changed
-      description: Support services have extraSelectorLabels
+      description: Added support for Gateway API Routes
```
Comment (bjw-s) [inline charts/library/common/Chart.yaml:17]: Please don't remove the "old" changelog for the `development` branch, but instead add to it.

**Review by bjw-s — CHANGES_REQUESTED** [review]
Thanks for submitting this PR! I have added a few comments on the chart.yaml. 

I am wondering if this is "enough" to fully support folks wanting to run the Gateway API? (as in, do we also need to somehow support the Gateway object?

**Comment by bjw-s** [comment]
Sorry about closing the PR, it did that automatically because of the branch being removed

File: charts/library/common/Chart.yaml (hunk 4 above)
Comment (andrasmaroy) [reply charts/library/common/Chart.yaml:5]: Weirdly enough I can't find an exact minimum version that the Gateway API needs, all I found is that they want to support the latest 5 minor versions. Personally I've been running it on 1.22.6 which is much newer than this, but I'll try to find something out about this.

File: charts/library/common/Chart.yaml (hunk 3 above)
Comment (andrasmaroy) [reply charts/library/common/Chart.yaml:6]: Weirdly enough I can't find an exact minimum version that the Gateway API needs, all I found is that they want to support the latest 5 minor versions. Personally I've been running it on 1.22.6 which is much newer than this, but I'll try to find something out about this.

File: charts/library/common/Chart.yaml (hunk 4 above)
Comment (andrasmaroy) [reply charts/library/common/Chart.yaml:5]: Alright, with the branch being reopened, is this still the case?

**Comment by andrasmaroy** [comment]
> Thanks for submitting this PR! I have added a few comments on the chart.yaml.
> 
> I am wondering if this is "enough" to fully support folks wanting to run the Gateway API? (as in, do we also need to somehow support the Gateway object?

Forgot to reply to this: to support using the Gateway API one needs to create a bunch of resources first (Gateway, GatewayClass, etc.) which are I feel like are out of the scope of this chart. To create routes that will actually connect the services through the Gateway API one needs to reference a Gateway resource for the route that is being created. I think this is a similar approach to how existing volume claims are handled.

I hope I noted this clearly enough in the documentation, but I'm more than happy to include more clarification if you feel like it needs more.

File: charts/library/common/Chart.yaml (hunk 5 above)
Comment (andrasmaroy) [reply charts/library/common/Chart.yaml:18]: If I figured correctly the current HEAD is released and everything is "new" so I ended up clearing the changes again, let me know if I misunderstood the situation.

**Comment by bjw-s** [comment]
I think you might need to rebase your changes off of the `development` branch again, looks like the history doesn't have all the commits/changes.

**Comment by andrasmaroy** [comment]
> I think you might need to rebase your changes off of the `development` branch again, looks like the history doesn't have all the commits/changes.

Should be good to  go now, cleaned up the commits a bit while at it.

**Review by bjw-s — APPROVED** [review]

**Pull request merged**